src/mlpack/methods/ann/layer/batch_norm.hpp

```diff
@@ -136,7 +142,13 @@ class BatchNorm
   OutputDataType TrainingMean() { return runningMean; }
 
   //! Get the variance over the training data.
-  OutputDataType TrainingVariance() { return runningVariance / count; }
+  OutputDataType TrainingVariance()
+  {
+    if (average)
+      return runningVariance / count;
```

Comment:
Because `count` is initialized with `0`, do you think it's good that we add a check that the value is `0` or not?

Reply:
I don't think user will call training variance before it's a single pass. It would be nice to add a check anyway.

Reply:
Agreed, so I was thinking it's good or not :)